pkg/utils/oci/helmregistry.go

```diff
@@ -52,6 +65,25 @@ func (r *HelmRegistry) Pull(ctx context.Context, oci *gardencorev1.OCIRepository
 		remote.WithContext(ctx),
 	}
 
+	if oci.PullSecretRef != nil {
+		namespace := v1beta1constants.GardenNamespace
+		if v := ctx.Value(ContextKeyPullSecretNamespace); v != nil {
+			s, ok := v.(string)
+			if !ok {
+				return nil, errors.New("pull secret namespace must be a string")
+			}
+			namespace = s
+		}
+		secret := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Namespace: namespace, Name: oci.PullSecretRef.Name}}
+		if err := r.client.Get(ctx, client.ObjectKeyFromObject(secret), secret); err != nil {
+			return nil, fmt.Errorf("failed to get pull secret %s: %w", client.ObjectKeyFromObject(secret), err)
+		}
+		if secret.Data == nil || secret.Data[corev1.DockerConfigJsonKey] == nil {
```

Comment:
```suggestion
		if secret.Data[corev1.DockerConfigJsonKey] == nil {
```

Nit. Checking for nil is not required.